test/test_c10d.py

```diff
@@ -3205,6 +3206,32 @@ def test_broadcast_coalesced_gloo_cpu(self):
         device = torch.device('cpu')
         self._test_broadcast_coalesced(process_group, device)
 
+class BackendLoadingTest(TestCase):
+    def _load_c10d_test_backend(self):
+        src = "{}/cpp_extensions/cpp_c10d_extension.cpp".format(os.path.abspath(os.path.dirname(__file__)))
+        include = "{}/../torch/lib".format(os.path.abspath(os.path.dirname(__file__)))
+        extension = load(
+            name="c10dtest",
+            sources=[src],
+            extra_include_paths=[include]
+        )
+
+    def test_collective_api(self):
+        self._load_c10d_test_backend()
+        os.environ['WORLD_SIZE'] = '1'
+        os.environ['MASTER_ADDR'] = '127.0.0.1'
+        os.environ['MASTER_PORT'] = str(common.find_free_port())
+        os.environ['RANK'] = '0'
+
+        c10d.init_process_group(backend='c10dtest', init_method='env://', world_size=1, rank=0)
+        self.assertEqual(c10d.get_rank(), 0)
+        self.assertEqual(c10d.get_world_size(), 1)
+        with tempfile.NamedTemporaryFile(delete=False) as file:
+            store = c10d.FileStore(file.name, 1)
+            process_group = c10d.distributed_c10d.ProcessGroupC10DTEST(store, 0, 1, '')
+
+            work = process_group.allreduce(torch.rand(1))
+            c10d.destroy_process_group()
```

Comment:
IIUC, this destroys the default group, but `c10d.distributed_c10d.ProcessGroupC10DTEST(store, 0, 1, '')` would create a new instance that's different from the default one?

Reply:
yes, you are right, I update the logic to use existing process group.